src/rgw/rgw_rest_s3.cc

```diff
@@ -103,7 +103,7 @@ int RGWGetObj_ObjStore_S3Website::send_response_data(bufferlist& bl, off_t bl_of
   iter = attrs.find(RGW_ATTR_AMZ_WEBSITE_REDIRECT_LOCATION);
   if (iter != attrs.end()) {
     bufferlist &bl = iter->second;
-    s->redirect = string(bl.c_str(), bl.length());
+    s->redirect = string(bl.c_str(), strlen(bl.c_str()));
```

Comment:
thanks. these null terminators in attributes keep on biting us :) see https://github.com/ceph/ceph/pull/15599 for a previous fix. it looks like this can be simplified to `s->redirect = bl.c_str();`

Reply:
@cbodley i have update the code，can this be  qa now?